models/league.go

```diff
@@ -0,0 +1,47 @@
+package models
+
+import (
+	"errors"
+
+	"github.com/google/uuid"
+	"github.com/gosimple/slug"
+)
+
+type League struct {
+	ID          string `bson:"_id,omitempty"`
+	Name        string `bson:"name,omitempty"`
+	Slug        string `bson:"slug,omitempty"`
+	Country     string `bson:"country,omitempty"`
+	CountrySlug string `bson:"country_slug,omitempty"`
+}
+
+type CreateLeaguePayload struct {
+	Name    string
+	Country string
+}
+
+func (l *League) SetSlug() error {
```

Comment:
Please create a constructor for a model. Remember - a model always should be in a valid state, if somebody forgot or didn't know about methods like this - the slug will be empty, the model in not valid state.